fairlearn/reductions/_moments/conditional_selection_rate.py

```diff
@@ -145,6 +157,49 @@ def load_data(self, X, y, **kwargs):
         """Load the specified data into the object."""
         super().load_data(X, y, event=_ALL, **kwargs)
 
+class EqualOpportunity(ConditionalSelectionRate):
+    r"""Implementation of Equalized Odds as a moment.
+
+    Adds conditioning on label compared to Demographic parity, i.e.
+
+    .. math::
+       P[h(X) = y' | A = a, Y = y] = P[h(X) = y' | Y = y] \; \forall a, y, y'
+
+    This implementation of :class:`ConditionalSelectionRate` defines
+    events corresponding to the unique values of the `Y` array.
+
+    The `prob_event` :class:`pandas:pandas.DataFrame` will record the
+    fraction of the samples corresponding to each unique value in
+    the `Y` array.
+
+    The `index` MultiIndex will have a number of entries equal to
+    the number of unique values for the sensitive feature, multiplied by
+    the number of unique values of the `Y` array, multiplied by two (for
+    the Lagrange multipliers for positive and negative constraints).
+
+    With these definitions, the :meth:`signed_weights` method
+    will calculate the costs according to Example 4 of
+    `Agarwal et al. (2018) <https://arxiv.org/abs/1803.02453>`_.
+    """
+
+    short_name = "EqualizedOdds"
+
+    def load_data(self, X, y, **kwargs):
+        """Load the specified data into the object."""
+        super().load_data(X, y,
+                          event=pd.Series(y).apply(lambda y: _LABEL + "=" + str(y)),
+                          **kwargs)
+
+class EqualOpportunity(ConditionalSelectionRate):
+    short_name = "EqualOpportunity"
+
+    def load_data(self, X, y, **kwargs):
+        """Load the specified data into the object."""
+        super().load_data(X, y,
+                          event=pd.Series(y).apply(lambda y: _LABEL + "=" + str(y)),
+                          filter_value=0,
```

Comment:
One small tweak: can we make the 'good value' a parameter to the constructor or this method? That might fold into @MiroDudik 's API review, though

Reply:
Yes, this was the original intention of the flag, but I also need the 'bad value' then so that they can be filtered out. But I think there is a workaround for that which I can do with some pandas API.  I will try it out. Thank you!

Reply:
I agree with Richard here. If you know the "good value" then the "bad value" is simply the other one (using some sort of unique/set magic to determine that). This may come in handy if/when we decide to move to multi-class classification, because suddenly it's not one or the other, but one of n, and then specifying the "good value" is way easier than specifying n-1 "bad values".

Reply:
I have made this change. Looks cleaner than the current implementation for sure, but I don't think that would work for the multi-class classification.

Reply:
There are likely more things that won't work for multi-class, but setting this up the right way to start with should prevent some future pains ;-)
Still says filter_value=0 above, though?

Reply:
Yes, this is the value that has to be filtered. Oh sorry, I think I misinterpreted the earlier comments then, Should I rename this parameter to `good_value`? I can do that!

Reply:
If we're going to call this true_positive_parity or similar, then `positive_value` might be better (also sounds less like a value judgement)

Reply:
Yeah, we could call this `positive_value`. I thought `filter_value` is generic and could be used for positive as well as, possibly, negative values too. Anyways, I can change it to `positive_value`. Let me know if I should.

Reply:
See my comment above. Simply enter `numpy.nan` into `event` where y=0 and keep `_LABEL + "=" + str(y)` where y=1.